rules/opentitan_gdb_test.bzl

```diff
@@ -0,0 +1,122 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+load("@lowrisc_opentitan//rules:rv.bzl", "rv_rule")
+
+def _opentitan_gdb_fpga_cw310_test(ctx):
+    test_script = '''#!/usr/bin/env bash
+        set -e
+
+        COLOR_RED='\\x1b[0;31m'
+        COLOR_GREEN='\\x1b[0;32m'
+        COLOR_PURPLE='\\x1b[0;35m'
+
+        function prefix_lines() {{
+            LABEL="$1"
+            COLOR="$2"
+            COLOR_RESET='\\x1b[m'
+            sed -Eu "s/(.*)/$COLOR[$LABEL]$COLOR_RESET \\1/"
+        }}
+
+        set -x
+
+        ./opentitantool load-bitstream --rom-kind=TestRom rom.bit
+
+        (openocd -f /usr/share/openocd/scripts/interface/ftdi/olimex-arm-usb-tiny-h.cfg \\
+            -c "adapter speed 500; transport select jtag; reset_config trst_and_srst" \\
+            -f {openocd_earlgrey_config} 2>&1 \\
+            | prefix_lines OPENOCD "$COLOR_PURPLE") &
+        OPENOCD_PID=$!
+
+        # For debugging, it may be useful to use `--init-command`, which causes
+        # GDB to drop to the interactive prompt when the script ends rather than
+        # exiting.
+        (/tools/riscv/bin/riscv32-unknown-elf-gdb --command=script.gdb 2>&1 \\
+            | prefix_lines GDB "$COLOR_GREEN") &
+
+        ./opentitantool console \\
+            --timeout 15s \\
```

Comment:
If I set this timeout to 0, or use a wrong `exit_success_string` (and use a smaller timeout like 3s in order not to wait) I get:

```
...
[CONSOLE] Exiting interactive console.
[CONSOLE] Error: Console timeout exceeded
+ kill 27634
+ wait
/workspace/bazel-cache/opentitan-sram-injection/7a4325616f8af64f213b635f80fd9a00/sandbox/processwrapper-sandbox/400/execroot/lowrisc_opentitan/bazel-out/k8-fastbuild-ST-2cc462681f62/bin/sw/device/examples/sram_program/sram_program_fpga_cw310_test.sh.runfiles/lowrisc_opentitan/sw/device/examples/sram_program/sram_program_fpga_cw310_test.sh: line 43: 27634 Terminated              ( openocd -f /usr/share/openocd/scripts/interface/ftdi/olimex-arm-usb-tiny-h.cfg -c "adapter speed 500; transport select jtag; reset_config trst_and_srst" -f util/openocd/target/lowrisc-earlgrey.cfg 2>&1 | prefix_lines OPENOCD "$COLOR_PURPLE" )
Target //sw/device/examples/sram_program:sram_program_fpga_cw310_test up-to-date:
  bazel-out/k8-fastbuild-ST-2cc462681f62/bin/sw/device/examples/sram_program/sram_program_fpga_cw310_test.sh
INFO: Elapsed time: 3.592s, Critical Path: 3.25s
INFO: 2 processes: 2 processwrapper-sandbox.
INFO: Build completed successfully, 2 total actions
//sw/device/examples/sram_program:sram_program_fpga_cw310_test           PASSED in 3.2s
```

Reply:
Thanks for pointing that out! I checked that `opentitantool console` does exit with a failure when it times out, so I was surprised that the script ignored it. It turns out that I've been masking failures with all this piping. I've now added `set -o pipefail` so the earlier non-zero exit codes are not ignored.